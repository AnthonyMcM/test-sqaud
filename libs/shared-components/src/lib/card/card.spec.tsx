import { render, screen } from '@testing-library/react';

import Card from './card';

describe('Card', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <Card make="BMW" model="1 Series" price="1000" reg="B24 XPC" />
    );
    // screen.debug();
    expect(baseElement).toMatchSnapshot();
  });

  it.only('Should display make and model', () => {
    const result = render(
      <Card make="BMW" model="1 Series" price="1000" reg="B24 XPC" />
    );

    const heading = result.getByText('BMW 1 Series');
    expect(heading).toBeTruthy();

    const reg = result.getByText('B24 XPC');
    expect(reg).toBeTruthy();

    const price = result.getAllByText('Price Â£2000');
    expect(price).toBeTruthy();
  });
});
