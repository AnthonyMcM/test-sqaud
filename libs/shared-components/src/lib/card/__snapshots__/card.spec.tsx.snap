// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render successfully 1`] = `
<body>
  <div>
    <div
      class="card__StyledCard-sc-1vtda9q-0 gsqpfj"
    >
      <div
        class="ch-card"
      >
        <div
          class="ch-card__content"
        >
          <h3>
            BMW 1 Series
          </h3>
          <p
            class="ch-mb--2"
          >
            B24 XPC
          </p>
          <p
            class="ch-mb--0"
          >
            <b>
              Price Â£1000
            </b>
          </p>
        </div>
      </div>
    </div>
  </div>
</body>
`;
